src/nix/ps.cc

```diff
@@ -0,0 +1,146 @@
+#include "nix/store/store-cast.hh"
+#include "nix/store/active-builds.hh"
+#include "nix/util/table.hh"
+#include "nix/util/terminal.hh"
+
+#include <nlohmann/json.hpp>
+
+using namespace nix;
+
+struct CmdPs : MixJSON, StoreCommand
+{
+    std::string description() override
+    {
+        return "list active builds";
+    }
+
+    Category category() override
+    {
+        return catUtility;
+    }
+
+    std::string doc() override
+    {
+        return
+#include "ps.md"
+            ;
+    }
+
+    void run(ref<Store> store) override
+    {
+        auto & tracker = require<QueryActiveBuildsStore>(*store);
+
+        auto builds = tracker.queryActiveBuilds();
+
+        if (json) {
+            printJSON(nlohmann::json(builds));
+            return;
+        }
+
+        if (builds.empty()) {
+            notice("No active builds.");
+            return;
+        }
+
+        /* Helper to format user info: show name if available, else UID */
+        auto formatUser = [](const UserInfo & user) -> std::string {
+            return user.name ? *user.name : std::to_string(user.uid);
+        };
+
+        Table table;
+
+        /* Add column headers. */
+        table.push_back({{"USER"}, {"PID"}, {"CPU", TableCell::Alignment::Right}, {"DERIVATION/COMMAND"}});
+
+        for (const auto & build : builds) {
+            /* Calculate CPU time - use cgroup stats if available, otherwise sum process times. */
+            std::chrono::microseconds cpuTime = build.utime && build.stime ? *build.utime + *build.stime : [&]() {
+                std::chrono::microseconds total{0};
+                for (const auto & process : build.processes)
+                    total += process.utime.value_or(std::chrono::microseconds(0))
+                             + process.stime.value_or(std::chrono::microseconds(0))
+                             + process.cutime.value_or(std::chrono::microseconds(0))
+                             + process.cstime.value_or(std::chrono::microseconds(0));
+                return total;
+            }();
+
+            /* Add build summary row. */
+            table.push_back(
+                {formatUser(build.mainUser),
+                 std::to_string(build.mainPid),
+                 {fmt("%.1fs",
+                      std::chrono::duration_cast<std::chrono::duration<float, std::chrono::seconds::period>>(cpuTime)
+                          .count()),
+                  TableCell::Alignment::Right},
+                 fmt(ANSI_BOLD "%s" ANSI_NORMAL " (wall=%ds)",
+                     store->printStorePath(build.derivation),
+                     time(nullptr) - build.startTime)});
+
+            if (build.processes.empty()) {
+                table.push_back(
+                    {formatUser(build.mainUser),
+                     std::to_string(build.mainPid),
+                     {"", TableCell::Alignment::Right},
+                     fmt("%s" ANSI_ITALIC "(no process info)" ANSI_NORMAL, treeLast)});
```

Comment:
For all of these `treeLast`, `treeConn`, `treeNull`, `treeLine`, can we add a space after this character please?

I'm always frustrated when I go to copy something from e.g. systemd's status output and get something like `├─304273` instead of `304273` because they didn't want to put a space between the tree character and its contents. Let's please not make this same mistake; I don't want to have to manually trim `└───bash`.

Reply:
Mm, okay. I would still like to see it changed, but if it'll be larger, then I'll leave it for now.